pyproject.toml

```diff
@@ -28,6 +28,7 @@ mistralai = { version = ">=0.1.6,<1.0.0", optional = true }
 groq = { version = ">=0.4.2,<1.0.0", optional = true }
 cohere = { version = "^5.2.5", optional = true }
 pinecone-client = { version = "^3.2.2", optional = true }
+astrapy = "^1.1.0"
```

Comment:
add astra as an optional dependency
```
poetry add --optional astrapy
```
Then update `[tool.poetry.extras]` with
```
astrapy = ["astrapy"]
```
and update `all` to include `astrapy`